File: easy/3754_concatenate_non_zero_digits_and_multiply_by_sum_i/solution.py
class Solution:
    def sumAndMultiply(self, n: int) -> int:
        digits = []

        while n:
            digit = n % 10

            if digit != 0:
                digits.append(digit)

            n //= 10

        digitsSum = sum(digits)

        while digits:
            n = n * 10 + digits.pop()

        return n * digitsSum

File: easy/3754_concatenate_non_zero_digits_and_multiply_by_sum_i/solution.cpp
class Solution {
public:
  long long sumAndMultiply(int n) {
    long long nNew = 0;
    int mul = 1;
    int digitsSum = 0;

    while (n) {
      int digit = n % 10;
      n /= 10;

      if (digit != 0) {
        nNew = digit * mul + nNew;
        mul *= 10;
        digitsSum += digit;
      }
    }

    return nNew * digitsSum;
  }
};
